tests/openthread/main.c

```diff
@@ -0,0 +1,213 @@
+#include <stdio.h>
+
+#include "net/ipv6/addr.h"
+#include "openthread/ip6.h"
+#include "openthread/thread.h"
+#include "openthread/udp.h"
+#include "ot.h"
+#include "shell.h"
+#include "shell_commands.h"
+
+otUdpSocket mSocket;
+
+static OT_JOB _set_panid(otInstance *ot_instance, void *context)
+{
+    uint16_t panid = *((uint16_t*) context);
+    otLinkSetPanId(ot_instance, panid);
+}
+
+static OT_JOB _get_panid(otInstance *ot_instance, void *context)
+{
+    *((uint16_t*) context) = otLinkGetPanId(ot_instance);
+    printf("PanID: %04x\n", *((uint16_t*) context));
+}
+
+static OT_JOB _thread_start(otInstance *ot_instance, void *context)
+{
+    printf("Starting OpenThread\n");
+    otIp6SetEnabled(ot_instance, true);
+    otThreadSetEnabled(ot_instance, true);
+}
+
+static OT_JOB _read_state(otInstance *ot_instance, void *context)
+{
+    uint8_t state = otThreadGetDeviceRole(ot_instance);
+
+	switch(state) {
+		 case kDeviceRoleOffline:
+            puts("offline");
+            break;
+        case kDeviceRoleDisabled:
+            puts("disabled");
+            break;
+        case kDeviceRoleDetached:
+            puts("detached");
+            break;
+        case kDeviceRoleChild:
+            puts("child");
+            break;
+        case kDeviceRoleRouter:
+            puts("router");
+            break;
+        case kDeviceRoleLeader:
+            puts("leader");
+            break;
+	}
+}
+
+static OT_JOB _get_ip_addresses(otInstance *ot_instance, void *context)
+{
+    for(const otNetifAddress *addr=otIp6GetUnicastAddresses(ot_instance); addr; addr=addr->mNext) {
+        char addrstr[IPV6_ADDR_MAX_STR_LEN];
+		ipv6_addr_to_str(addrstr, (ipv6_addr_t*) &addr->mAddress.mFields, sizeof(addrstr));
+        printf("inet6 %s\n", addrstr);
+    }
+}
+
+void _handle_receive(void *aContext, otMessage *aMessage, const otMessageInfo *aMessageInfo)
+{
+    size_t payload_len = otMessageGetLength(aMessage)-otMessageGetOffset(aMessage);
+
+    char buf[100];
+    otMessageRead(aMessage, otMessageGetOffset(aMessage), buf, payload_len);
+
+    printf("Message: ");
+    for(int i=0;i<payload_len;i++) {
+        printf("%02x ", buf[i]);
+    }
+    printf("\n");
+}
+
+static OT_JOB _create_udp_socket(otInstance *ot_instance, void *context)
+{
+    otSockAddr sockaddr;
+    memset(&sockaddr, 0, sizeof(otSockAddr));
+    sockaddr.mPort = *((uint16_t*) context);
+
+    otUdpOpen(ot_instance, &mSocket, _handle_receive, NULL);
+    otUdpBind(&mSocket, &sockaddr);
+}
+
+typedef struct 
+{
+    ipv6_addr_t ip_addr;
+    uint16_t port;
+    void *payload;
+    size_t len;
+} udp_pkt_t;
+
+static OT_JOB _send_udp_pkt(otInstance *ot_instance, void *context)
+{
+    udp_pkt_t *pkt = (udp_pkt_t*) context;
+    otMessage *message;
+
+    otUdpSocket socket;
+    memset(&socket, 0, sizeof(otUdpSocket));
+
+    otUdpOpen(ot_instance, &socket, _handle_receive, NULL);
+
+    message = otUdpNewMessage(ot_instance, true);
+    int error;
```

Comment:
you don't need error since you void it 3 lines before. Simply do:
```c
otMessageSetLength(message, pkt->len);
otMessageWrite(message, 0, pkt->payload, pkt->len);
```